study.go

```diff
@@ -201,6 +205,69 @@ func (s *Study) GetSystemAttrs() (map[string]string, error) {
 	return s.Storage.GetStudySystemAttrs(s.ID)
 }
 
+// ToInTrialStudy converts Study to InTrialStudy.
+func ToInTrialStudy(study *Study) *InTrialStudy {
+	return &InTrialStudy{
+		ID:      study.ID,
+		Name:    study.Name,
+		Sampler: study.Sampler,
```

Comment:
To replace `trial.Study` with InTrialStudy, we need to provide Sampler from InTrialStudy for `trial.SuggestXXX` methods. It might be unclear because InTrialStudy is also passed sampler interface (Pruner attribute is also unclear because of same reasons).

Reply:
seems to be removed at https://github.com/pfnet/optuna/pull/491